edb/server/dbview/dbview.pyx

```diff
@@ -1124,6 +1080,42 @@ cdef class DatabaseConnectionView:
                 if not lock._waiters:
                     del lock_table[query_req]
 
+        recompiled_cache = None
```

Comment:
Should this logic go in a separate function? (Fine if you don't think so)

Reply:
Yeah, I was thinking about having all this decided in `server/compiler/compiler.py` or deeper where we can have per-statement "dbview" of session_config, database_config and all (it's like what `dbview.pyx` does in `start()`/`on_success()`, but it'll be too late for recompilation), so that we can handle scripts like:

```
configure session set auto_rebuild_query_cache := false;
create type Foo;
configure session set auto_rebuild_query_cache := true;
```

or use the correct config to recompile after scripts like:

```
configure current database set apply_access_policies := true;
create Type Foo;
```

So basically this logic is temporarily here and I'm expecting it to be replaced by a simple `if query_unit_group.should_recompile` or something like that in the near future.